pygmt/modules.py

```diff
@@ -214,3 +217,85 @@ def __exit__(self, exc_type, exc_value, traceback):
         )
         with Session() as lib:
             lib.call_module("set", arg_str)
+
+
+@xr.register_dataarray_accessor("gmt")
+class GMTDataArrayAccessor:
+    """
+    This is the GMT extension for :class:`xarray.DataArray`.
+
+    You can access various GMT specific metadata as follow:
+
+    >>> from pygmt.datasets import load_earth_relief
+    >>> # Use the global Earth relief grid with 1 degree spacing
+    >>> grid = load_earth_relief(resolution="01d")
+
+    >>> # See if grid uses Gridline (0) or Pixel (1) registration
+    >>> grid.gmt.registration
+    0
+    >>> # See if grid uses Cartesian (0) or Geographic (1) coordinate system
+    >>> grid.gmt.gtype
+    1
+    """
+
+    def __init__(self, xarray_obj):
+        self._obj = xarray_obj
+        try:
+            self._source = self._obj.encoding["source"]
+            self._info = grdinfo(self._source)
+        except KeyError:
+            default_reg_and_gtype = "Gridline node registration used [Cartesian grid]"
+            # logging.warning(
+            #    msg="Cannot find a NetCDF source of xarray grid. "
+            #    f"Will fallback to using GMT's default setting: {default_reg_and_gtype}"
+            # )
+            self._info = default_reg_and_gtype
+
+    @property
+    def registration(self):
+        """
+        Registration type of the grid, either Gridline (0) or Pixel (1).
+        """
+        try:
+            return self._registration
+        except AttributeError:
+            if "Gridline node registration used" in self._info:
+                self._registration = 0
+            elif "Pixel node registration used" in self._info:
+                self._registration = 1
```

Comment:
Can definitely be refactored once we hard pin to `GMT > 6.1.0`, though I do like the readability here.

Reply:
Ok, refactored in 896ed590ae5ce5198c1af810e729f4159436f1a8.